test/functional/feature_segwit.py

```diff
@@ -279,6 +280,34 @@ def run_test(self):
         # Mine a block to clear the gbt cache again.
         self.nodes[0].generate(1)
 
+        self.log.info("Signing with all-segwit inputs reveals fee rate")
+        # Test that signing a tx with all-segwit inputs returns the fee
+        # in the results
+        addr = self.nodes[0].getnewaddress(address_type='p2sh-segwit')
+        txid = self.nodes[0].sendtoaddress(addr, 1)
+        tx = self.nodes[0].getrawtransaction(txid, True)
+        n = -1
+        value = -1
+        for o in tx["vout"]:
+            if o["scriptPubKey"]["addresses"][0] == addr:
+                n = o["n"]
+                value = Decimal(o["value"])
+                break
+        assert n > -1 # failure means we could not find the address in the outputs despite sending to it
```

Comment:
This can be a one-liner: `output = next(o for o in tx["vout"] if o["scriptPubKey"]["addresses"][0] == addr)`

Reply:
Maybe, but that's a bit unreadable?